biocrnpyler/sbmlutil.py

```diff
@@ -90,10 +105,10 @@ def add_all_species(model, species: List, initial_condition_dictionary: dict, co
                 initial_condition_dictionary[s])
         else:
             initial_concentration = 0
-        add_species(model=model, compartment=compartment,
-                    species=s, initial_concentration=initial_concentration)
+        add_species(model=model, compartment=compartment, species_name=str(s.name), 
+                            species_id=s_id, initial_concentration=initial_concentration)
 
-def add_species(model, compartment, species, initial_concentration=None, **kwargs):
+def add_species(model, compartment, species_name, species_id, initial_concentration=None,  **kwargs):
```

Comment:
I was wondering why this one is better than figuring out the `species_id` inside `add_species()`?

Reply:
because it has to check through all the ids in the model already. this way it only checks them all once instead of every time you add a species

Reply:
got it!